### PR TITLE
PRMT-4467

### PR DESCRIPTION
**Pull request by martin-nhs**
- Removed downloading the file from S3 in favour of `get_object`.
- Added additional validation.

### CODE REVIEW

**Review by chrisbloe-nhse — APPROVED** [review]
This looks good.

File: lambda/orc_ingestion.py [hunk 1]
```diff
@@ -58,27 +60,43 @@ def lambda_handler(event, context):
...
+    response = s3_client.get_object(
+        Bucket=bucket_name,
+        Key=filename
     )
-    with open(local_file_path, "r") as f:
-        nhs_number_list = f.read()
-    os.remove(local_file_path)
-    return nhs_number_list.split(',')
+
+    file_contents = response["Body"].readlines()
```
Comment (martin-nhs) [inline lambda/orc_ingestion.py:68]:
```suggestion
    file_lines = response["Body"].readlines()
```

File: lambda/orc_ingestion.py [hunk 2]
```diff
@@ -58,27 +60,43 @@ def lambda_handler(event, context):
...
+
+    if len(file_contents) > 1:
+        raise InvalidFileFormatException(
+            "All NHS numbers must be contained on a single line, seperated by commas."
+        )
```
Comment (martin-nhs) [inline lambda/orc_ingestion.py:73]:
```suggestion
        raise InvalidFileFormatException("All NHS numbers must be contained on a single line, separated by commas.")
```

File: lambda/orc_ingestion.py [hunk 3]
```diff
@@ -58,27 +60,43 @@ def lambda_handler(event, context):
...
-    with open(local_file_path, "r") as f:
-        nhs_number_list = f.read()
-    os.remove(local_file_path)
-    return nhs_number_list.split(',')
+
+    file_contents = response["Body"].readlines()
+
+    if len(file_contents) > 1:
+        raise InvalidFileFormatException(
+            "All NHS numbers must be contained on a single line, seperated by commas."
```
Comment (JoeSykesNHS) [inline lambda/orc_ingestion.py:72]: separated

File: lambda/orc_ingestion.py [hunk 4]
```diff
@@ -58,27 +60,43 @@ def lambda_handler(event, context):
...
+        )
+
+    nhs_numbers_str = file_contents[0].decode('utf-8')
+
+    only_nums_and_commas_bool = bool(re.compile(r'[0-9,]').match(nhs_numbers_str))
```
Comment (JoeSykesNHS) [inline lambda/orc_ingestion.py:77]: could we test to see if this line works when the re.complie is removed as I think the compile may be done by the match.

suggested code:
only_nums_and_commas_bool = bool(re.match('^[0-9,]+$', nhs_numbers_str))

Comment (JoeSykesNHS) [reply lambda/orc_ingestion.py:77]: regex from this python example

image

**Review by JoeSykesNHS — APPROVED** [review]

File: lambda/orc_ingestion.py [hunk 5]
```diff
@@ -15,116 +16,88 @@
...
 
+class S3FileReader:
+    @staticmethod
+    def read(filename, bucket_name) -> str:
+        response = s3_client.get_object(Bucket=bucket_name, Key=filename)
+        file_lines = response["Body"].readlines()
 
-def new_uuid() -> str:
-    return str(uuid.uuid4())
+        if len(file_lines) > 1:
```
Comment (JoeSykesNHS) [inline lambda/orc_ingestion.py:53]: bit pedantic so feel free to ignore:

I would separate this validation out into a separate function under the class and then calls Class.read() and then class.validate_one_line()

**Review by JoeSykesNHS — APPROVED** [review]

**Pull request merged**